index 4b47351..3835103 100644
--- FinalStateAnalysis/Utilities/interface/ExpressionNtupleColumn.h
+++ FinalStateAnalysis/Utilities/interface/ExpressionNtupleColumn.h
@@ -69,7 +69,7 @@ class ExpressionNtupleColumnT : public ExpressionNtupleColumn<ObjType> {
       return new ExpressionNtupleColumnT(name, func, tree);
     }
     catch(cms::Exception& iException){
-      iException.addContext("Cought exception in building branch: " + name + " with formula: " + func);
+      iException << "Cought exception in building branch: " + name + " with formula: " + func;
       throw;
     }
   }
